vars/piperStageWrapper.groovy

```diff
@@ -29,17 +29,19 @@ void call(Map parameters = [:], body) {
         .use()
 
     stageLocking(config) {
-        def containerMap = ContainerMap.instance.getMap().get(stageName) ?: [:]
-        if (Boolean.valueOf(env.ON_K8S) && (containerMap.size() > 0 || config.runStageInPod)) {
-            withEnv(["POD_NAME=${stageName}"]) {
-                dockerExecuteOnKubernetes(script: script, containerMap: containerMap, stageName: stageName) {
+        withEnv(["STAGE_NAME=${stageName}"]) {
```

Comment:
@OliverNocon @CCFenner Would be good to get your feedback on this change (adding withEnv(["STAGE_NAME=${stageName}"])), if you see any issue with it. 
Only with this change to go steps will be able to resolve the stage config if the technical identifier is used.

Reply:
not sure how Jenkins behaves since we would basically change an environment variable which is set by Jenkins.
Maybe I am too paranoid ...

Another option would be to pass the stage name to the `piper` binary as parameter in case the names differ ...

Don't we then still have the problem that we cannot support `id` and `name` of the stage in parallel?
Teams would need to decide which way to go `id` or `name`. Is this an issue?
Should we strive for a solution where `id` and `name` can be used interchangeably?

Reply:
Don't know if this contributes meaningfully to the discussion, but I've tested the following pipeline:

```groovy
pipeline {
    agent any
    stages {
        stage('Init') {
            steps {
                echo "Stage name in Init: ${env.STAGE_NAME}"
                withEnv(["STAGE_NAME=Test"]) {
                    echo "Stage name inside withEnv: ${env.STAGE_NAME}"
                }
                echo "Stage name after withEnv: ${env.STAGE_NAME}"
            }
        }
        stage('Build and Test') {
            steps {
                echo "Stage name in build and test: ${env.STAGE_NAME}"
            }
        }
    }
}
```

It gives the following output:

```
[Pipeline] Start of Pipeline
[Pipeline] node
Running on agent01 in /var/jenkins_home/workspace/sandbox-pipeline
[Pipeline] {
[Pipeline] stage
[Pipeline] { (Init)
[Pipeline] echo
Stage name in Init: Init
[Pipeline] withEnv
[Pipeline] {
[Pipeline] echo
Stage name inside withEnv: Test
[Pipeline] }
[Pipeline] // withEnv
[Pipeline] echo
Stage name after withEnv: Init
[Pipeline] }
[Pipeline] // stage
[Pipeline] stage
[Pipeline] { (Build and Test)
[Pipeline] echo
Stage name in build and test: Build and Test
[Pipeline] }
[Pipeline] // stage
[Pipeline] }
[Pipeline] // node
[Pipeline] End of Pipeline
Finished: SUCCESS
```

Reply:
We also do it here https://github.com/SAP/jenkins-library/blob/ded6152b56d6dc691a36a449f1dc87f45398d694/vars/piperPipelineStageArtifactDeployment.groovy#L32
have not seen any issues yet.

Reply:
@OliverNocon I changed to to a less invasive approach where I only set the env variable when this step is called with a todo to improve the stage config resolution to also support both kind if stage names.